src/pycram/designator.py

```diff
@@ -679,6 +688,44 @@ def __repr__(self):
                 [f"{f.name}={self.__getattribute__(f.name)}" for f in dataclasses.fields(self)] + [
                     f"pose={self.pose}"]) + ')'
 
+        def special_knowledge_adjustment_pose(self, grasp, pose):
+            """
+            Returns the adjusted target pose based on special knowledge for "grasp front".
+            """
+
```

Comment:
You could also to the transform to object frame here to ensure that the pose is in the correct frame.